lms/djangoapps/verify_student/views.py

```diff
@@ -376,6 +377,19 @@ def get(
         # so we can fire an analytics event upon payment.
         request.session['attempting_upgrade'] = (message == self.UPGRADE_MSG)
 
+        # Determine the course verification status
+        course_enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+        all_course_modes, _ = CourseMode.all_and_unexpired_modes_for_courses([course_key])
+        verification_good_until = ''
+        if course_enrollment and all_course_modes:
+            verify_status_by_course = check_verify_status_by_course(
+                request.user,
+                [(course, course_enrollment)],
+                all_course_modes
+            )
+            if verify_status_by_course:
+                verification_good_until = verify_status_by_course[course_key].get('verification_good_until') or ''
```

Comment:
I think this is more complicated than what you need.  Why not just retrieve the most recent software secure photo verification for the course with status either "submitted" or "approved", then use the date from that?

We're already making a similar query in `_check_already_verified` -- we should need only one database query to check whether (a) the user has a submitted or approved verification attempt, and (b) the expiration date of that attempt.